### PR TITLE
feat(dev_env): Install newer packages for Python 3.8

### PR DESCRIPTION
**Pull request by armenzg**
This allows installing newer packages on Python 3.8 in order to support Apple M1 development.

### CODE REVIEW

File: requirements-base.txt [hunk 1]
```diff
@@ -73,7 +74,8 @@ kombu==4.6.11
 # Note, grpcio>1.30.0 requires setting GRPC_POLL_STRATEGY=epoll1
 # See https://github.com/grpc/grpc/issues/23796 and
 # https://github.com/grpc/grpc/blob/v1.35.x/doc/core/grpc-polling-engines.md#polling-engine-implementations-in-grpc
-grpcio==1.35.0
+grpcio==1.35.0; python_version == '3.6'
+grpcio==1.39.0; python_version > '3.6'
```
Comment (EvanPurkhiser) [inline requirements-base.txt:78]: Is there any reason we can't just try and upgrade some of these? These don't _seem_ like they would be terrible right?

Comment (armenzg) [reply requirements-base.txt:78]: I didn't want to deal with testing these packages at this moment in order to move quickly.
IMHO, it's OK to diverge temporarily this dev env vs M1's dev env.
I can add it to my post-completion bucket if you wish to.

**Review by evanpurkhiser — APPROVED** [review]

File: requirements-base.txt (hunk 1 above)
Comment (joshuarli) [reply requirements-base.txt:78]: @armenzg I can deal with upgrading grpcio.

Comment (armenzg) [reply requirements-base.txt:78]: thank you @joshuarli !

**Pull request merged**